services/venice-controller/src/main/java/com/linkedin/venice/ingestion/control/RealTimeTopicSwitcher.java

```diff
@@ -222,7 +222,31 @@ public void transmitVersionSwapMessage(Store store, int previousVersion, int nex
 
     // If there exists an RT, then broadcast the Version Swap message to it, otherwise broadcast it to the VT
     String storeName = store.getName();
-    if (!topicManager.containsTopic(pubSubTopicRepository.getTopic(Version.composeRealTimeTopic(storeName)))) {
+    String rtForPreviousVersion = Utils.getRealTimeTopicName(previousStoreVersion);
+    String rtForNextVersion = Utils.getRealTimeTopicName(nextStoreVersion);
+    boolean rtExistsForPreviousVersion = previousStoreVersion.isHybrid()
+        && topicManager.containsTopic(pubSubTopicRepository.getTopic(rtForPreviousVersion));
+    boolean rtExistsForNextVersion =
+        nextStoreVersion.isHybrid() && topicManager.containsTopic(pubSubTopicRepository.getTopic(rtForNextVersion));
+
+    if (rtExistsForPreviousVersion || rtExistsForNextVersion) {
+      if (rtExistsForPreviousVersion) {
+        LOGGER.info(
+            "RT topic exists for store: {}, versionNum: {}. Broadcasting Version Swap message directly to RT.",
+            storeName,
+            previousVersion);
```

Comment:
Nit, for the ease of future debugging, could we also log `previousStoreVersion`, `nextStoreVersion`, and `rtForPreviousVersion`?

Reply:
added rt topic name in logs. logging version number instead of vesion, because versionimpl.tostring() prints too much detail of a version